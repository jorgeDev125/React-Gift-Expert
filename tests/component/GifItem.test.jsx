import { render, screen } from '@testing-library/react';
import { GifItem } from '../../src/components';


describe('Pruebas en el componente <GifItem/>', () => {

    const title = 'Dragon Ball';
    const url = 'http://localhost/https//dragonball.com/goku.png';

        test('deberÃ­a hacer match con el snapshot', () => {

            const { container } = render ( <GifItem title = { title } url={ url } /> );
            expect ( container ).toMatchSnapshot();
        });

        test('Debe de mostrar la imagen con el URL y el ALT indicado', () => {

            render ( <GifItem title = { title } url={ url } /> );
            // screen.debug();
            // expect( screen.getByRole('img').src).toBe(url);
            // expect( screen.getByRole('alt').src).toBe(title);
            const { src, alt } = screen.getByRole('img');
            expect ( src ).toBe( url );
            expect ( alt ).toBe ( title );
           
        });

        test('debe de mostrar que el titulo existe', () => {

            render ( <GifItem title = { title } url={ url } /> );
            expect ( screen.getByText( title ) ).toBeTruthy();
        });
});

