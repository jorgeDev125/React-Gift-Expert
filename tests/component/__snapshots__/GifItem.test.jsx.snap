// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <GifItem/> deber√≠a hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Dragon Ball"
      src="http://localhost/https//dragonball.com/goku.png"
    />
    <p>
      Dragon Ball
    </p>
  </div>
</div>
`;
